lua/toggleterm/terminal.lua

```diff
@@ -175,6 +169,9 @@ end
 function Terminal:resize(size)
   if self:is_split() then
     ui.resize_split(self, size)
+    if config.get("persist_size") then
+      ui.save_direction_size(size, self.direction)
```

Comment:
This still leaks persistence implementation details and also doesn't look necessary anymore since resize split now already calls `ui.save_direction_size` right? the `resurrect_only` argument can just be renamed to `save_size` or something using a named args table and set to `true` for session restore as well as terminal resizing

Reply:
But it is necessary to check whether `persistent[term.direction]` is `nil` to determine whether it is really session restore.

Reply:
I'm not sure I follow, there isn't a place inside this function where that check is being done though, maybe what I was getting at isn't clear.

I just mean calling save direction inside of the terminal means that it still references stuff to do with persistence but when I read through the implementation of `resize_split` it can already call `save_direction_size` if the `resurrect_only` argument is true so since it is already being called in resize split then why call it again here rather than just call resize split with the third arg as true.

But in this case it's confusing to call it `resurrect_only` it should just be called `save_size` or something. Does that make sense?

Reply:
I guess I need to change `Terminal:__resurrect()` to eliminate the need for `resurrect_only`.

Reply:
@zeertzjq checking the config value is fine but I think maybe what I mean isn't clear. I've explained a bit more below. It could be that I'm missing something though